src/contracts/core/StakeRootCompendium.sol

```diff
@@ -26,170 +25,139 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
 
     /// @inheritdoc IStakeRootCompendium
     function depositForOperatorSet(IAVSDirectory.OperatorSet calldata operatorSet) external payable {
-        require(
-            avsDirectory.isOperatorSet(operatorSet.avs, operatorSet.operatorSetId),
-            "StakeRootCompendium.depositForOperatorSet: operator set does not exist"
-        );
-        depositBalanceInfo[operatorSet.avs][operatorSet.operatorSetId].balance += msg.value;
-        require(
-            depositBalanceInfo[operatorSet.avs][operatorSet.operatorSetId].balance >= 2 * MIN_DEPOSIT_BALANCE,
-            "StakeRootCompendium.depositForOperatorSet: depositer must have at least 2x the minimum balance on deposit"
-        );
-
-        // update the deposit balance for the operator set whenever a deposit is made
-        _updateDepositBalanceInfo(operatorSet, true);
+        _deposit(operatorSet);
     }
 
     /// @inheritdoc IStakeRootCompendium
     function addStrategiesAndMultipliers(
         uint32 operatorSetId,
         StrategyAndMultiplier[] calldata strategiesAndMultipliers
     ) external {
-        require(
-            strategiesAndMultipliers.length > 0,
-            "StakeRootCompendium.setStrategiesAndMultipliers: no strategies and multipliers provided"
-        );
+        _addStrategiesAndMultipliers(IAVSDirectory.OperatorSet({avs: msg.sender, operatorSetId: operatorSetId}), strategiesAndMultipliers);
+    }
+
+    // natspec
+    /// @notice Initializes an operator set with a given set of strategies and multipliers
+    /// @dev This a convinience (gas saving) function for avs'
+    /// @param operatorSetId The ID of the operator set to initialize
+    /// @param strategiesAndMultipliers The strategies and multipliers to initialize the operator set with
+    function initOperatorSetInStakeTree(
```

Comment:
hm.

Can we just make `addStrategiesAndMultipliers` payable and deposit if `msg.value > 0`?